### ISSUE TITLE
Trailing Space in a comment before the code to be fixed adds unnecessary empty line

### ISSUE THREAD

**Issue by dbaltas** [FriendsOfPHP/PHP-CS-Fixer#3457]
Having some code with a potential fix and a comment with a trailing space right before the code, results to an empty line being added in the fix.
If the comment has no trailing space, fixer works as expected.

#### The PHP version you are using (`$ php -v`):
=> PHP 7.1.8

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> 2.10.0

#### The command you use to run PHP CS Fixer:
=> `php-cs-fixer fix a.php`

#### Sample PHP code (a.php):
 * before running PHP CS Fixer (no changes):
```php
<?php

function foo() : void
{
    // there is a trailing space in this comment! 
    $var1 = (
        $var2 ?? 0);
}

```

 * with unexpected changes applied when running PHP CS Fixer:
```php
<?php

function foo() : void
{
    // there is a trailing space in this comment!
    $var1 = (
        $var2 ?? 0
 
    );
}

```
> In the resulted code above, there is an empty line between `$var2 ?? 0` and the closing parenthesis

 * with the changes you expected instead:
```php
<?php

function foo() : void
{
    // there is a trailing space in this comment!
    $var1 = (
        $var2 ?? 0
    );
}
```

**Comment by SpacePossum** [comment]
Hi and thanks for reporting,

Could you share your configuration? Using the default configuration of this project on v.2.10.0 does not show the issue so it is hard to track down without knowing which rules are applied, thanks!

**Comment by dbaltas** [comment]
Is it possible that the issue is OS dependent? I am running it on MacOS 10.13.2.
As you can see below the issue on my computer is reproducable without any configuration file. (`Loaded config default.`)
```
11:55:13 ➜  /tmp  uname -a
Darwin dimitris-mbp 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64
11:55:19 ➜  /tmp  php-cs-fixer fix --dry-run --diff-format=udiff a.php
Loaded config default.
   1) a.php
      ---------- begin diff ----------
--- Original
+++ New
@@ -2,7 +2,9 @@

 function foo() : void
 {
-    // there is a trailing space in this comment! 
+    // there is a trailing space in this comment!
     $var1 = (
-        $var2 ?? 0);
+        $var2 ?? 0
+
+    );
 }

      ----------- end diff -----------


Checked all files in 0.009 seconds, 8.000 MB memory used
11:55:23 ➜  /tmp
```

**Comment by dbaltas** [comment]
This is my a.php file
```
curl https://gist.githubusercontent.com/dbaltas/54c06eee04113c1e4ab1df2edde06a9b/raw/265d31685eeb4f6d65d189c97fb1ee1e55aabe5a/trailing-space-in-comment-3457 > a.php
```
I got the unnecessary blank line on Linux as well.

**Comment by dbaltas** [comment]
Here is a test reproducing the issue https://github.com/dbaltas/PHP-CS-Fixer/commit/b4ef7ea8f5db265b9bd4f972112811ca6651a099

**Comment by keradus** [comment]
maybe just raise a PR with it ?

**Comment by dbaltas** [comment]
Well, I didn't raise a PR because I am not close to suggest a fix.

**Comment by SpacePossum** [comment]
I retried this on `2.2`, `2.10` and `master` and cannot reproduce this issue anymore.
@dbaltas my check might not be all complete, maybe I didn't setup the failing cases correctly so please; if you've the time please retry with any of the latest stable released versions (2.*) of the fixer and let me know if this is still an issue, if so I reopen it :)  thanks for reporting!

**Comment by dbaltas** [comment]
The issue persists in 2.12.1
### steps to reproduce
```
rm -rf /tmp/cs-fix
mkdir /tmp/cs-fix
cd /tmp/cs-fix
curl https://gist.githubusercontent.com/dbaltas/18befe7f6582b10168557a0fcdcc234b/raw/18df7964f384b84aabb2ac0649fb8dbd1bd254a9/trailing-space-in-comma-3547.php > a.php
cat a.php
php-cs-fixer --version
php-cs-fixer fix .
cat a.php
```

### logs
```
5:24:50 ➜  /tmp  rm -rf /tmp/cs-fix
5:24:57 ➜  /tmp  mkdir /tmp/cs-fix
5:25:06 ➜  /tmp  cd /tmp/cs-fix
5:25:11 ➜  cs-fix  curl https://gist.githubusercontent.com/dbaltas/18befe7f6582b10168557a0fcdcc234b/raw/18df7964f384b84aabb2ac0649fb8dbd1bd254a9/trailing-space-in-comma-3547.php > a.php
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   119  100   119    0     0    182      0 --:--:-- --:--:-- --:--:--   182
5:25:20 ➜  cs-fix  cat a.php
<?php

function foo() : void
{
    // there is a trailing space in this comment!
    $var1 = (
        $var2 ?? 0);
}
5:25:30 ➜  cs-fix  php-cs-fixer --version
PHP CS Fixer 2.12.1 Long Journey by Fabien Potencier and Dariusz Ruminski
5:25:44 ➜  cs-fix  php-cs-fixer fix .
Loaded config default.
   1) a.php

Fixed all files in 0.008 seconds, 10.000 MB memory used
5:25:53 ➜  cs-fix  cat a.php
<?php

function foo() : void
{
    // there is a trailing space in this comment!
    $var1 = (
        $var2 ?? 0

    );
}
5:25:57 ➜  cs-fix
```
Above is the terminal log of creating a file /tmp/cs-fix/a.php with "the before running" version.
After running `php-cs-fixer fix`, a new empty line has been added after the `$var2 ?? 0`.

I wouldn't expect the operating system to affect this. (I am running on MacOS 10.13.4, php 7.1.8)

**Comment by SpacePossum** [comment]
Thanks for getting back!
Seems this is indeed still an issue, my apologies for closing it early :(

After digging around this seems a priority issue between;
`method_argument_space` and `no_trailing_whitespace_in_comment`,

When `no_trailing_whitespace_in_comment` is run before `method_argument_space` it goes wrong (these both have priority 0)

However this does _not_ happen when not using the rules default configuration;
something like this goes ok:
`$ php php-cs-fixer fix tmp/test2.php -vvv --dry-run --diff --rules=method_argument_space,no_trailing_whitespace`

but it does go wrong for something like this fails (note: no config file used);
`$ php php-cs-fixer fix tmp/test2.php -vvv --dry-run --diff`

so the priority issue is somewhere between the two rules _and_ their configuration... bug-hunt continues!
